### PR TITLE
Feature/mifareclassic readsector

### PR DESCRIPTION
**Pull request by poison**
## Added
- introduced mifareClassicReadBlock
- added mifareClassicGetBlockCountInSector, mifareClassicGetSectorCount and mifareClassicSectorToBlock

## Changed
- renamed mifareClassicReadBlock (that was reading a whole sector) to mifareClassicReadSector

Note to reviewer: please check commits individually, will be better to follow

### CODE REVIEW

File: android/src/main/java/community/revteltech/nfc/NfcManager.java [hunk 1]
```diff
@@ -859,20 +966,23 @@ private void writeNdef(Tag tag, WriteNdefRequest request) {
 		}
 	}
 
-	private byte[] rnArrayToBytes(ReadableArray rArray) {
+	private static byte[] rnArrayToBytes(ReadableArray rArray) {
```
Comment (whitedogg13) [inline android/src/main/java/community/revteltech/nfc/NfcManager.java]: cool, thanks to make these util method as `static`

File: android/src/main/java/community/revteltech/nfc/NfcManager.java [hunk 2]
```diff
@@ -859,20 +966,23 @@ private void writeNdef(Tag tag, WriteNdefRequest request) {
...
 			bytes[i] = (byte)(rArray.getInt(i) & 0xff);
 		}
 		return bytes;
 	}
 
-	public WritableArray bytesToRnArray(byte[] bytes) {
-        WritableArray value = Arguments.createArray();
-        for (int i = 0; i < bytes.length; i++) {
-            value.pushInt((bytes[i] & 0xFF));
+	public static WritableArray bytesToRnArray(byte[] bytes) {
```
Comment (whitedogg13) [inline android/src/main/java/community/revteltech/nfc/NfcManager.java]: Just grep, no other place use this method, can we make it `private`?

File: android/src/main/java/community/revteltech/nfc/NfcManager.java [hunk 3]
```diff
@@ -859,20 +966,23 @@ private void writeNdef(Tag tag, WriteNdefRequest request) {
...
+		return appendBytesToRnArray(Arguments.createArray(), bytes);
+	}
+
+	public static WritableArray appendBytesToRnArray(WritableArray value, byte[] bytes) {
```
Comment (whitedogg13) [inline android/src/main/java/community/revteltech/nfc/NfcManager.java]: How about also make it `private`?

**Comment by poison** [comment]
Allright, done, let me know what you think!

**Comment by whitedogg13** [comment]
Thanks a lot!

**Pull request merged**